### ISSUE TITLE
Collate1D tolerance TypeError

### ISSUE THREAD

**Issue by woodml** [pypeit/PypeIt#1810]
When using pypeit_collate1d, I encountered a type error if the input tolerance is an integer. 
The error occurred both when tolerance was set in an input file and when it was set using the command line.

The documentation example (https://pypeit.readthedocs.io/en/1.14.0/collate1d.html) said that the tolerance should be a float or integer, depending on match type:
"For ra/dec matching, this is an angular distance. For pixel matching it's an integer or floating point number."

Example:
Input: 
```pypeit_collate_1d  20240108.collate1d --tolerance 3```
```
File "~/anaconda3/envs/pypeit/lib/python3.11/site-packages/pypeit/par/parset.py", line 231, in __setitem__
    raise TypeError('Input value for {0} has incorrect type: {1}.'.format(key, value) +
TypeError: Input value for tolerance has incorrect type: 3.
Valid types are: [<class 'str'>, <class 'float'>]
```

Input file parameters:
```
# User-defined coadding and fluxing parameters can be given but are not required
[coadd1d]
sn_clip = 20

[fluxcalib]
extinct_correct = True

# User-defined collating parameters
[collate1d]
	match_using = pixel
	tolerance = 3
	#slit_exclude_flags = BOXSLIT
	exclude_serendip = False
	flux = False
	refframe = heliocentric
	ignore_flux = False
	wv_rms_thresh = 0.2
	outdir = Science/
	#chk_version = False
```

**Comment by badpandabear** [comment]
Plan to fix this in PR https://github.com/pypeit/PypeIt/pull/1814

**Comment by kbwestfall** [comment]
Fixed in #1814
